### PR TITLE
Swagger changes for object lifecycle management

### PR DESCRIPTION
**Pull request by nguptaopensds**
**What this PR does / why we need it**:
These are swagger changes for lifecycle feature

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
```

### CODE REVIEW

File: openapi-spec/swagger.yaml [hunk 1]
```diff
@@ -1052,6 +1122,56 @@ components:
...
+                description: 'storage class where you want to transition.'
+                type: string
+              Backend:
+                description: 'backend where you wish to transition.'
+                type: string
+          Expiration:
+            description: 'expiration action in lifecycle rule.'
+            type: object
+            properties:
+              Days:
```
Comment (leonwanghui) [inline openapi-spec/swagger.yaml]: s/Days/days

File: openapi-spec/swagger.yaml [hunk 2]
```diff
@@ -679,6 +679,76 @@ paths:
...
+          required: true
+          schema:
+            type: string
+      tags:
+        - s3
+      summary: create a bucket lifecycle.
+      operationId: bucketLifecyclePut
+      responses:
+        '200':
+          description: successful create bucket lifecycle
```
Comment (kumarashit) [inline openapi-spec/swagger.yaml]: successfully created bucket lifecycle

File: openapi-spec/swagger.yaml [hunk 3]
```diff
@@ -1052,6 +1122,56 @@ components:
...
+          Status:
+            description: 'the status of lifecycle rule whether its enabled or disabled.'
+            type: string
+            example: ENABLE
+          Transition:
+            description: 'transition action in lifecycle rule.'
+            type: object
+            properties:
+              Days:
+                description: 'days after creation of object when you wish to transition.'
```
Comment (kumarashit) [inline openapi-spec/swagger.yaml]:
```suggestion
                description: 'days after creation of object when user wish to transition.'
                 
```

File: openapi-spec/swagger.yaml [hunk 4]
```diff
@@ -679,6 +679,76 @@ paths:
...
+      parameters:
+        - name: bucketName
+          in: path
+          description: the name of bucket
+          required: true
+          schema:
+            type: string
+        - name: ruleID
+          in: query
+          description: the rule id in lifecycle configuration of the bucket, used in delete call
```
Comment (sfzeng) [inline openapi-spec/swagger.yaml]: Please give the format if it is multiple ruleIDs.

**Comment by nguptaopensds** [comment]
Closing this PR as the context have been changed in new release

**Pull request closed without merging**